lib/set.js

```diff
@@ -4,17 +4,18 @@ var format = require('util').format
 
 var Promise = require('lie')
 
-var generateSalt = require('./utils/generate-salt')
-var hashPassword = require('./utils/hash-password')
+var internals = module.exports.internals = {}
+internals.hashPassword = require('./utils/hash-password')
+internals.generateSalt = require('./utils/generate-salt')
```

Comment:
This is what I meant with exposing `internals`. This way you you do

```js
var set = require('./path/to/set')
```

then set.internals.hashPassword is accessible. That way we can mock its behavior in tests